common/aggregate_functions/src/aggregate_combinator_distinct.rs

```diff
@@ -120,14 +143,30 @@ impl IAggregateFunction for AggregateUniqFunction {
     }
 
     fn merge_result(&self) -> Result<DataValue> {
-        Ok(DataValue::UInt64(Some(self.state.len() as u64)))
+        // faster path for count
+        if self
+            .nested
+            .as_any()
+            .downcast_ref::<AggregateCountFunction>()
+            .is_some()
+        {
+            Ok(DataValue::UInt64(Some(self.state.len() as u64)))
+        } else {
+            // accumulate_scalar
+            let mut nested = self.nested.clone();
+            self.state.iter().try_for_each(|group_values| {
```

Comment:
We will call accumulate_scalar when merge stage,  standalone mode, this will be single threaded, which may be slow.

we need to create an issue to track and improve it

Reply:
noted a to-do comment.

Reply:
It's even single-threaded in ClickHouse implementation.